Makefile

```diff
@@ -309,37 +309,41 @@ setup-develop: venv $(KAT_CLIENT) $(TELEPROXY) $(KUBERNAUT) version
 
 kill_teleproxy = $(shell kill -INT $$(/bin/ps -ef | fgrep venv/bin/teleproxy | fgrep -v grep | awk '{ print $$2 }') 2>/dev/null)
 
-cluster.yaml: $(CLAIM_FILE)
+cluster.yaml: $(CLAIM_FILE) kubernaut-claim teleproxy-restart
+	rm -rf /tmp/k8s-*.yaml
+
+kubernaut-claim:
 ifeq ($(USE_KUBERNAUT), true)
 	$(KUBERNAUT_DISCARD)
 	$(KUBERNAUT_CLAIM)
 	cp ~/.kube/$(CLAIM_NAME).yaml cluster.yaml
 endif
-	rm -rf /tmp/k8s-*.yaml
-	@echo "Killing teleproxy"
-	$(call kill_teleproxy)
-	$(TELEPROXY) -kubeconfig $(KUBECONFIG) 2> /tmp/teleproxy.log || (echo "failed to start teleproxy"; cat /tmp/teleproxy.log) &
-	@echo "Sleeping for Teleproxy cluster"
-	sleep 10
 
 setup-test: cluster.yaml
 
-teleproxy-restart:
-	@echo "Killing teleproxy"
-	$(call kill_teleproxy)
-	sleep 0.25 # wait for exit...
+teleproxy-start:
+	@echo "Starting teleproxy"
 	@$(TELEPROXY) -kubeconfig $(KUBECONFIG) 2> /tmp/teleproxy.log || (echo "failed to start teleproxy"; cat /tmp/teleproxy.log) &
-	@echo "Done"
+	@echo "Waiting for teleproxy to start..."
+	@sleep 5
+	@if ! pgrep -x "teleproxy" > /dev/null; then \
+		echo "Teleproxy crashed, exiting..."; \
+		exit 1; \
+	fi
+	@echo "Started teleproxy"
+
+teleproxy-restart: teleproxy-stop teleproxy-start
```

Comment:
These will run in arbitrary order

Reply:
Hmm. Are you sure? I've been running this target quite a bit and it has been working for me just fine. :confused:

Reply:
Which "these" do you mean, @ark3?

Reply:
Stop and start *may* run in arbitrary order.

Reply:
Dealt with in `master` and merged in here.